components/dialog/dialog-mixin.js

```diff
@@ -260,6 +262,11 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 		this._focusFirst();
 	}
 
+	_handleFullscreenWithin(e) {
+		if (e.detail.state) this._fullscreenWithin += 1;
```

Comment:
This code weakly keeps track of whether or not there are fullscreen elements within.  It assumes that those descendants will only dispatch the event once per state.  If this proves unreliable, we can track references to the nodes themselves.  I don't expect this to be used very often - the editor might end up being the only case.